docs/trace-command.md

```diff
@@ -0,0 +1,117 @@
+```
+
+### State of a resource
+
+Each object in the trace output categorized in one of the following states which is represented in text (state column of
+overview) and graph output (as node attributes) accordingly.
+
+| State | Text Representation | Graph Representation | 
+| :--------: | :---: | :---: |
+| Ready    | ✔ | Solid - Black |
+| NotReady | ! | Solid - Orange |
+| Missing  | ✕ | Dotted - Red |
+
+### Sample outputs
+
+#### Text output 
+
+![alt text](media/trace-text-example.png "Trace Text output example")
+
+#### Graph output 
+
+![alt text](media/trace-graph-example.png "Trace Graph output example")
+
+
+## How it works
+
+The trace tool categorizes Crossplane resources based on a mapping between resource group/kinds and one of the following 
+Crossplane resource types. Then it finds the related resources with the help of the references available in the resource
+declarations as shown below. 
+
+- [Kubernetes Application](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#kubernetes-application)
+    - `status.clusterRef` -> KubernetesCluster
+    - listing/filtering all `KubernetesApplicationResources` and `ResourceClaims` in the same namespace with labels 
+  at `spec.resourceSelector.matchLabels`
+  
+- Kubernetes Application Resource 
+    - `status.clusterRef` -> KubernetesCluster
+    - `spec.secrets` -> [Connection secrets](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#connection-secrets)
+   
+- [Resource Claim](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#kubernetes-application)
+    - `spec.resourceRef` -> Managed Resource
+    - `spec.classRef` -> Resource Class
+    - `spec.writeConnectionSecretToRef` -> [Connection secrets](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#connection-secrets)
+
+- [Managed Resource](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#managed-resource)
+    - `spec.claimRef` -> Resource Claim
+    - `spec.classRef` -> Resource Class
+    - `spec.writeConnectionSecretToRef` -> [Connection secrets](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#connection-secrets)
+  
+- [Resource Class](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#non-portable-resource-class) 
+    - `specTemplate.providerRef` -> Provider
+
+- [Provider](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#provider)
+    - `spec.credentialsSecretRef` -> Credentials Secret
+    
+Knowing related objects for a given one, the trace tool builds a graph representing the relations. The data of the 
+graph is outputted either in text format or as [graph definition dot format](https://www.graphviz.org/doc/info/lang.html). 
+Text output contains both an overview of all related types indicating [their state](#state-of-a-resource))
+
+## Future Work
```

Comment:
Nice - I like that we included this section to give people a preview of what we have in mind